rust/crd/src/lib.rs

```diff
@@ -273,7 +294,30 @@ impl SparkApplication {
         let mode = self.mode().context(ObjectHasNoDeployModeSnafu)?;
         let name = self.metadata.name.clone().context(ObjectHasNoNameSnafu)?;
 
-        let mut submit_cmd = vec![
+        let mut submit_cmd: Vec<String> = vec![];
+
+        let s3_conn = s3bucket.as_ref().and_then(|i| i.connection.as_ref());
+
+        if let Some(S3ConnectionSpec {
+            credentials: Some(_credentials),
+            ..
+        }) = s3_conn
+        {
+            submit_cmd.push(format!(
+                "export {env_var}=$(cat {secret_dir}/{file_name}) && ",
```

Comment:
I understood why it's necessary to read the credentials from a file. But I don't understand why we need to pass them through the environment anymore.